pkg/plugin/client/client.go

```diff
@@ -38,10 +42,42 @@ type CustomCommandExecutor interface {
 	ExecuteCustomCommand(cmdName string, ctx context.Context, args []string) error
 }
 
-type clientFactory struct{}
+type clientFactory struct {
+	bzl bazel.Bazel
+}
+
+// buildPlugin asks bazel to build the target and returns the path to the resulting binary
+func (c *clientFactory) buildPlugin(target string) (string, error) {
+	agc, err := c.bzl.AQuery(target)
+	if err != nil {
+		return "", err
+	}
+	outs := bazel.ParseOutputs(agc)
+
+	var pluginPath string
+	for _, a := range outs {
+		// TODO: don't hard-code GoLink, plugins could be written in other languages
```

Comment:
Could this be a GitHub issue instead so we can keep track and actually schedule work to address this?